docs/project-decisions/0002-single-responsibility-per-file.md

```diff
@@ -0,0 +1,24 @@
+# 1. Record architecture decisions
+
+Date: 2021-12-13
+
+## Status
+
+Accepted
+
+## Context
+
+We want clean code. Part of that is to limit a code file to a single responsibility.
+
+## Decision
+
+Each code file (including but not limited to `.php`, `.js`, `.ts`, `.sh`) should have a single responsibility.
+As a consequence of that decision:
+- Interfaces, Types, Classes and global functions should get their own file.
```

Comment:
This doesn't match the [original conclusion](https://yoast.atlassian.net/browse/AR-122?focusedCommentId=27861), where types may be exported without being the default export, and thus not having its own file. Was this intentional?

Reply:
@diedexx good point - I was a Scrooge with words.

Reply:
@diedexx how do you like the current version?